### PR TITLE
Add no_str and no_arr options for ros2 topic echo

### LINKED ISSUE #136: [ros2topic] implement `--noarr` and `--nostr` like in ROS 1
#### Feature description

`echo` verb:
- provide a `--noarr` argument that will replace array field with `<array type: <FIELD_TYPE>, length: <ARRAY_LENGTH>>`
- provide a `--nostr` argument that will replace string field with `<string length: <STRING_LENGTH>>`

This will allow to improve performance of printing long messages as well as not relying on a default arbitrary truncation (like we currently do).

### PR DESCRIPTION
**Pull request by vinnamkim**
Resolves #136, depends on https://github.com/ros2/rosidl_python/pull/38

Signed-off-by: vinnamkim <vinnam.kim@gmail.com>

### CODE REVIEW

File: ros2topic/ros2topic/verb/echo.py [hunk 1]
```diff
@@ -64,16 +64,22 @@ def add_arguments(self, parser, cli_name):
             '--truncate-length', '-l', type=unsigned_int, default=DEFAULT_TRUNCATE_LENGTH,
             help='The length to truncate arrays, bytes, and string to '
                  '(default: %d)' % DEFAULT_TRUNCATE_LENGTH)
+        parser.add_argument(
+            '--noarr', action='store_true', help='Exclude arrays')
```
Comment (emersonknapp) [inline ros2topic/ros2topic/verb/echo.py]: Nitpicks:
* for more clarity, should the argument be `--no-arr`?
* `help='Don't print array fields of messages.'`

File: ros2topic/ros2topic/verb/echo.py [hunk 2]
```diff
@@ -64,16 +64,22 @@ def add_arguments(self, parser, cli_name):
...
+        parser.add_argument(
+            '--nostr', action='store_true', help='Exclude string fields')
```
Comment (emersonknapp) [inline ros2topic/ros2topic/verb/echo.py]: Nitpicks:
* for more clarity, should the argument be `--no-str`?
* `help='Don't print string fields of messages.'`

File: ros2topic/ros2topic/verb/echo.py (hunk 1 above)
Comment (vinnamkim) [reply ros2topic/ros2topic/verb/echo.py]: Fixed.

File: ros2topic/ros2topic/verb/echo.py (hunk 2 above)
Comment (vinnamkim) [reply ros2topic/ros2topic/verb/echo.py]: Fixed.

**Comment by vinnamkim** [comment]
@emersonknapp updated PR.

**Comment by jacobperron** [comment]
CI: https://github.com/ros2/rosidl_python/pull/38#issuecomment-509469677

File: ros2topic/ros2topic/verb/echo.py [hunk 3]
```diff
@@ -112,15 +118,15 @@ def subscriber(node, topic_name, message_type, callback):
         rclpy.spin_once(node)
 
 
-def subscriber_cb(truncate_length):
+def subscriber_cb(truncate_length, noarr, nostr):
     def cb(msg):
-        nonlocal truncate_length
-        print(message_to_yaml(msg, truncate_length), end='---\n')
+        nonlocal truncate_length, noarr, nostr
+        print(message_to_yaml(msg, truncate_length, noarr, nostr), end='---\n')
```
Comment (dirk-thomas) [inline ros2topic/ros2topic/verb/echo.py]: Please pass keyword arguments explicitly with the keyword.

Same below.

Comment (dirk-thomas) [reply ros2topic/ros2topic/verb/echo.py]: @jacobperron Friendly ping.

Comment (jacobperron) [reply ros2topic/ros2topic/verb/echo.py]: https://github.com/ros2/ros2cli/pull/317

**Pull request merged**